pkg/functions/riff/builder.go

```diff
@@ -0,0 +1,180 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package riff
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/functions"
+	"github.com/vmware/dispatch/pkg/trace"
+)
+
+func (d *riffDriver) Shutdown() {
+	defer trace.Trace("")()
+	defer func() { recover() }() // close can panic
+	close(d.buildRequests)
+}
+
+func cleanup(tmpDir string) {
+	defer trace.Tracef("rm dir: %s", tmpDir)()
+	os.RemoveAll(tmpDir)
+}
+
+func (d *riffDriver) processRequests() {
+	for r := range d.buildRequests {
+		r.result <- d.buildAndPushImage(r)
+		close(r.result)
+	}
+}
+
+func dockerError(r io.ReadCloser, err error) error {
```

Comment:
This function is copied from openfaas/builder.go.  Can we just make it common?

Reply:
yeah, making the builder a common util is on my list